### PR TITLE
fix stdcm display get

### LINKED ISSUE #5589: Operational points names are wrong in STDCM
### What happened?

Those labels in space-time chart are wrong in stdcm, they all display `stop-...` :

image

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

1. Run STDCM

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: W11
- **Env**: dev

### OSRD version (top right corner `Account` button > `Informations`)

c39e6996

### PR DESCRIPTION
**Pull request by younesschrifi**
close #5589

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java [hunk 1]
```diff
@@ -47,6 +49,7 @@ STDCMResult makeResult(
         var mergedEnvelopes = STDCMUtils.mergeEnvelopeRanges(ranges);
         var departureTime = computeDepartureTime(ranges, startTime);
         var stops = makeStops(ranges);
+        stops.addAll(makeOpStops(infra, trainPath));
```
Comment (Erashin) [inline core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java]: Is there a possibility of there being duplicates in this list ? When `makeOperationalWaypoints` is called in `PathfindingResultConverter.makePathWaypoint`, the combination of the two lists passes through a call of `sortAndMergeDuplicates`.

**Review by clarani — COMMENTED** [review]
LGTM and tested (for the front) :white_check_mark:

**Review by Wadjetz — APPROVED** [review]

File: core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java [hunk 2]
```diff
@@ -160,4 +163,41 @@ private static List<TrainStop> makeStops(List<Pathfinding.EdgeRange<STDCMEdge>>
...
+        return res;
+    }
+
+    /** Sorts the stops on the path. When stops overlap, the user-defined one is kept. */
+    private static List<TrainStop> sortAndMergeStopsDuplicates(List<TrainStop> stops) {
+        stops.sort(Comparator.comparingDouble(st -> st.position));
+        var res = new ArrayList<TrainStop>();
+        TrainStop last = null;
+        for (var stop : stops) {
+            if (last != null && Math.abs(last.position - stop.position) < 0.001)
```
Comment (Erashin) [inline core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java]: When comparing positions, we wanna use `arePositionsEqual()` from now on. See `TrainPhysicsIntegrator.java`.

File: core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java [hunk 3]
```diff
@@ -160,4 +163,41 @@ private static List<TrainStop> makeStops(List<Pathfinding.EdgeRange<STDCMEdge>>
...
+                last.position = stop.position;
+            else {
+                last = stop;
+                res.add(last);
+            }
+        }
+        return res;
+    }
+
+    /** Make the list of stops on the path, in order. Both user-defined stops and operational points. */
```
Comment (Erashin) [inline core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java]: `Make the path's ordered list of stops. Contains both user-defined...`

File: core/src/test/kotlin/fr/sncf/osrd/utils/DummyInfra.kt [hunk 4]
```diff
@@ -420,7 +420,7 @@ class DummyInfra : RawInfra, BlockInfra {
     }
 
     override fun getTrackChunkOperationalPointParts(trackChunk: TrackChunkId): StaticIdxList<OperationalPointPart> {
-        TODO("Not yet implemented")
+        return mutableStaticIdxArrayListOf()
```
Comment (Erashin) [inline core/src/test/kotlin/fr/sncf/osrd/utils/DummyInfra.kt:423]: Don't know what the plan is with `DummyInfra`. Might wanna check with a maintainer if this should be really implemented and added to test suite.

File: core/src/main/java/fr/sncf/osrd/api/stdcm/STDCMEndpoint.java [hunk 5]
```diff
@@ -134,14 +134,13 @@ private static List<STDCMStep> parseSteps(FullInfra infra, List<STDCMRequest.STD
...
 
-    /** Generate a train schedule matching the envelope and rolling stock, with one stop at the end */
+    /** Generate a train schedule matching the envelope and rolling stock */
     public static StandaloneTrainSchedule makeTrainSchedule(
             double endPos,
             RollingStock rollingStock,
             RollingStock.Comfort comfort,
             List<TrainStop> trainStops
     ) {
-        trainStops.add(new TrainStop(endPos, 0.1));
```
Comment (eckter) [inline core/src/main/java/fr/sncf/osrd/api/stdcm/STDCMEndpoint.java:144]: Wouldn't that cause the train to end its path with a non-zero speed?

**Review by Erashin — APPROVED** [review]
LGTM, modulo a couple of comments.

File: core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java [hunk 6]
```diff
@@ -46,7 +49,7 @@ STDCMResult makeResult(
         var physicsPath = EnvelopeTrainPath.from(infra, trainPath);
         var mergedEnvelopes = STDCMUtils.mergeEnvelopeRanges(ranges);
         var departureTime = computeDepartureTime(ranges, startTime);
-        var stops = makeStops(ranges);
+        var stops = makePathStops(ranges, infra, trainPath);
```
Comment (eckter) [inline core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.java]: This changes how the standard allowance is computed, the allowance ranges don't span across stops.

In this module all stops are considered to be places the train actually stops at, the operational points shouldn't be involved

**Review by eckter — COMMENTED** [review]
The changes shouldn't affect the STDCM module, there can be dangerous side-effects. In this module the stops are assumed to be places the train actually stops at.

I'm a little confused about how it fixes the problem and how the responses are handled. I don't think the `stop-*` prefixes are added by core. There is already a list of operational points in the response, which should be identical to the ones generated in normal pathfindings.

File: core/src/test/kotlin/fr/sncf/osrd/utils/DummyInfra.kt (hunk 4 above)
Comment (younesschrifi) [reply core/src/test/kotlin/fr/sncf/osrd/utils/DummyInfra.kt:423]: I talked about that with @eckter.

**Review by eckter — APPROVED** [review]

**Review by clarani — APPROVED** [review]
Oups, forget to approve :white_check_mark:

**Pull request merged**